### PR TITLE
XWIKI-21091: AbstractDocumentsMigration is not resolving the document references against the current wiki

### PR DESCRIPTION
**Pull request by manuelleduc**
Jira: https://jira.xwiki.org/browse/XWIKI-21091

### CODE REVIEW

File: xwiki-platform-core/xwiki-platform-index/xwiki-platform-index-default/src/main/java/org/xwiki/index/migration/R140300000XWIKI19614DataMigration.java [hunk 1]
```diff
@@ -48,6 +49,7 @@
 @Singleton
 @Named(R140300000XWIKI19614DataMigration.HINT)
 @Deprecated(since = "14.8RC1")
+@Unstable
```
Comment (surli) [inline xwiki-platform-core/xwiki-platform-index/xwiki-platform-index-default/src/main/java/org/xwiki/index/migration/R140300000XWIKI19614DataMigration.java:52]: It's internal you don't need unstable here.

Comment (surli) [inline xwiki-platform-core/xwiki-platform-index/xwiki-platform-index-default/src/main/java/org/xwiki/index/migration/R140300000XWIKI19614DataMigration.java:52]: Should have been internal no?

File: xwiki-platform-core/pom.xml [hunk 2]
```diff
@@ -226,6 +226,19 @@
...
+                  <ignore>true</ignore>
+                  <code>java.annotation.added</code>
+                  <old>field org.xwiki.internal.migration.AbstractDocumentsMigration.documentReferenceResolver @ org.xwiki.index.migration.R140300000XWIKI19614DataMigration</old>
+                  <new>field org.xwiki.internal.migration.AbstractDocumentsMigration.documentReferenceResolver @ org.xwiki.index.migration.R140300000XWIKI19614DataMigration</new>
+                  <annotation>@javax.inject.Named("current")</annotation>
+                  <justification>Young API</justification>
+                  <criticality>documented</criticality>
+                </item>
+              </differences>
+            </revapi.differences>
```
[the remark this answers is not in the document]
Comment (surli) [reply xwiki-platform-core/pom.xml:241]: I'm not sure why migration would be exposed API tbh

File: xwiki-platform-core/pom.xml [hunk 3]
```diff
@@ -141,6 +141,17 @@
...
               </differences>
             </revapi.differences>
+            <revapi.differences>
+              <differences>
+                <item>
+                  <criticality>highlight</criticality>
+                  <ignore>true</ignore>
+                  <code>java.class.removed</code>
+                  <old>class org.xwiki.index.migration.R140300000XWIKI19614DataMigration</old>
+                  <justification>Moved to legacy where this class should have been from the beginning.</justification>
```
Comment (surli) [inline xwiki-platform-core/pom.xml:151]: s/legacy/internal no?

Comment (manuelleduc) [reply xwiki-platform-core/pom.xml]: yes... thanks, fixed.

File: xwiki-platform-core/xwiki-platform-index/xwiki-platform-index-default/src/main/java/org/xwiki/index/internal/migration/R140300000XWIKI19614DataMigration.java [hunk 4]
```diff
@@ -70,7 +70,7 @@ public String getDescription()
     @Override
     public XWikiDBVersion getVersion()
     {
-        return new XWikiDBVersion(140900000);
+        return new XWikiDBVersion(150700000);
```
[the remark this answers is not in the document]
Comment (manuelleduc) [reply xwiki-platform-core/xwiki-platform-index/xwiki-platform-index-default/src/main/java/org/xwiki/index/internal/migration/R140300000XWIKI19614DataMigration.java:74]: Indeed. Done.

**Review by surli — APPROVED** [review]
LGTM

**Pull request merged**